### DISCUSSION TITLE
DCMotor for motors with rotation encoders

### DISCUSSION THREAD

**Discussion by Novakasa** [pybricks/support#1364] [Q&A] [answered]
Hey all,

Instantiating a `DCMotor` for a motor with rotation encoder used to work in earlier betas of 3.3. It seems that with current master, this now triggers a `OSError: [Errno 19] ENODEV`. I have seen some maybe related discussion in https://github.com/pybricks/support/issues/1162, but this in the end seemed to only be about the `PUPDevice`. Is this a regression or was the compatibility of `DCMotor` with the motors with encoders unintended behaviour from the beginning?

Example program (with a 88008 medium motor on Port A)
```python
from pybricks.hubs import CityHub
from pybricks.pupdevices import DCMotor, Light
from pybricks.parameters import Port
from pybricks.tools import wait

hub = CityHub()
motor = DCMotor(port=Port.A)
motor.dc(100)

wait(1000)
```

**Comment by laurensvalk** [comment, marked as answer]
> Is this a regression or was the compatibility of DCMotor with the motors with encoders unintended behaviour from the beginning?

The latter. Each class is intended to be used with a specific class of devices.

It should be easy enough to revisit this though. Is there a use case that can't be achieved with try/except?

**Reply by Novakasa** [reply]
No, this is easily accounted for in my project, I just wanted to make sure before committing to the change. Thanks!
